codebase2/core/U/Codebase/HashTags.hs

```diff
@@ -2,6 +2,10 @@ module U.Codebase.HashTags where
 
 import Unison.Hash (Hash)
 
+-- | Represents a hash of a type or term component
+newtype ComponentHash = ComponentHash {unComponentHash :: Hash}
```

Comment:
I'm surprised we didn't have a tag for this already, there are a lot of spots we should likely be using it. I added something like this into Enlil for my pg name-lookup stuff, but it should really just be part of unison core.